R/fertilityChecks.R

```diff
@@ -85,92 +85,64 @@ checkFertilityEvent_latlon <- function(latitude
-      }
+    cat(paste0('    Requesting Data between ',currentStartDate,' and ',currentEndDate,'\n'))
 
+    suppressWarnings(returnedData[[length(returnedData) +1]] <- as.data.table(aWhereAPI::daily_observed_latlng(latitude
+                                                                                                              ,longitude
+                                                                                                              ,currentStartDate
+                                                                                                              ,currentEndDate
+                                                                                                              ,propertiesToInclude = c('temperatures'))))
 
-      if (currentStartDate <= (Sys.Date()-1)) {
-        seqDateLength_2 <- length(seq(lubridate::ymd(currentStartDate),lubridate::ymd(currentEndDate),1))
 
-        cat(paste0('     Requesting Data between ',currentStartDate,' and ',currentEndDate,'\n'))
 
-        suppressWarnings(returnedData[[length(returnedData) +1]] <- as.data.table(aWhereAPI::daily_observed_latlng(latitude
-                                                                                                                   ,longitude
-                                                                                                                   ,currentStartDate
-                                                                                                                   ,currentEndDate
-                                                                                                                   ,propertiesToInclude = c('temperatures'))))
+    returnedData[[length(returnedData)]][,seqDatePosition := seq(1,.N,1)]
 
-        returnedData[[length(returnedData)]][,seqDatePosition := seq(seqDateLength_1+1,seqDateLength_1 + seqDateLength_2,1)]
-      }
-    }
+    returnedData[[length(returnedData)]][,paste0(c('temperatures.max_roll_'
+                                                   ,'temperatures.min_roll_')
+                                                 ,numConsecutiveDaysToCheck) := list(round(zoo::rollapply(data = temperatures.max
+                                                                                                          ,width = numConsecutiveDaysToCheck
+                                                                                                          ,FUN = mean
+                                                                                                          ,na.rm = TRUE
+                                                                                                          ,align = 'right'
+                                                                                                          ,fill = NA),2)
+                                                                                     ,round(zoo::rollapply(data = temperatures.max
```

Comment:
Should the data in this line be temperatures.min?

Reply:
Fixed, good catch